### PR TITLE
[Az.RecoveryServices.Backup] Adding Cross Zonal Restore for Zone-pinned VMs

### PR DESCRIPTION
**Pull request by hiaga**
## Description

Adding Cross Zonal Restore for Zone-pinned VMs

## Checklist

- [Yes] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [Yes] The title of the PR is clear and informative
- [Yes] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [Yes] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [N/A] If applicable, the changes made in the PR have proper test coverage
- [Yes] For public API changes to cmdlets:
    - [Yes] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [Yes] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by VeryEarly** [comment]
looks like this SDK upgrade have API version upgrade involved, if so, test cases need to be re-recorded.

**Comment by hiaga** [comment]
@VeryEarly : How do I know what all test cases are needing to be rerecorded?

**Comment by hiaga** [comment]
@wyunchi-ms : could you pls review this PR

**Comment by hiaga** [comment]
cmdlet review pr: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/810

File: src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Restore/RestoreAzureRMRecoveryServicesBackupItem.cs [hunk 1]
```diff
@@ -223,6 +222,15 @@ public class RestoreAzureRmRecoveryServicesBackupItem : RSBackupVaultCmdletBase
...
         public SwitchParameter RestoreToSecondaryRegion { get; set; }
 
+        /// <summary>
+        /// Target Zone Number to restore the VM disks
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = AzureVMParameterSet,
+            HelpMessage = ParamHelpMsgs.RestoreVM.TargetZone)]
+        [Parameter(Mandatory = false, ParameterSetName = AzureVMManagedDiskParameterSet,
+            HelpMessage = ParamHelpMsgs.RestoreVM.TargetZone)]
+        public string TargetZoneNumber { get; set; }
```
Comment (wyunchi-ms) [inline src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Restore/RestoreAzureRMRecoveryServicesBackupItem.cs:232]: Need to be integer.

Comment (hiaga) [reply src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Restore/RestoreAzureRMRecoveryServicesBackupItem.cs]: Done.

**Comment by hiaga** [comment]
@wyunchi-ms : changed the type from string to int. Pls review

**Review by wyunchi-ms — APPROVED** [review]

**Pull request merged**